### PR TITLE
Disable kubectl validation for upgrade e2e

### PR DESCRIPTION
**Pull request by nikhiljindal**
Fixes https://github.com/kubernetes/kubernetes/issues/22884

cc @ihmccreery @bgrant0607

### CODE REVIEW

**Comment by bgrant0607** [comment]
The kubectl flag looks ok. I don't see other examples of code looking at testContext.RepoRoot, though.

@ixdy @ihmccreery does that seem ok?

**Comment by ihmccreery** [comment]
Use `serverVersionGTE`, (in `e2e/util.go`).

**Comment by nikhiljindal** [comment]
thanks, updating now

**Comment by nikhiljindal** [comment]
Pushed a new version that uses serverVersionGTE/
PTAL

File: test/e2e/kubectl.go [hunk 1]
```diff
@@ -101,6 +101,12 @@ var (
 	//
 	// TODO(ihmccreery): remove once we don't care about v1.1 anymore, (tentatively in v1.4).
 	deploymentsVersion = version.MustParse("v1.2.0-alpha.7.726")
+
+	// Pod probe parameters were introduced in #15967 (v1.2) so we dont expect tests that use
+	// these probe parameters to work on clusters before that.
+	//
+	// TODO(ihmccreery): remove once we don't care about v1.0 and v1.1 anymore, (tentatively in v1.4).
```
Comment (ihmccreery) [inline test/e2e/kubectl.go]: Remove once we don't care about v1.1, right?

**Comment by ihmccreery** [comment]
LGTM other than comment.

File: test/e2e/kubectl.go (hunk 1 above)
Comment (nikhiljindal) [inline test/e2e/kubectl.go]: Fixed

**Comment by nikhiljindal** [comment]
Fixed the TODO and pushed a new version. PTAL

**Comment by ihmccreery** [comment]
LGTM, thanks!

**Comment by nikhiljindal** [comment]
cc @k8s-oncall for hand merge since this is 1.2 P0

**Pull request merged**